### DISCUSSION TITLE
Songchange plugin - Command to run on stopping a song

### DISCUSSION THREAD

**Discussion by AbraxasDMG** [audacious-media-player/audacious#115] [Ideas]
Hi everyone.

Could you please cover one more event with songchange plugin?

Essentially, what is missing is the ability to run a command when deliberately stopping the song. That could be covered by extending the existing "Command to run at the end of a song" event or add a new one (with dedicated text box) for "Command to run at stopping the song".

I'll try to explain this request with my own use case scenario.
I use a program that reads a text file and displays whatever is in it via OSD.
When the song changes (from one file to another on a local drive, or on the radio), the title of the running song is updated in the OSD, because I update the text file from Audiacious.

When the song from a local file ends, and if it is the last one in the list, Audacious executes the external command and the text file is emptied out, which also reflects in OSD. There are no songs currently running and OSD is properly showing that.
For radio stations it is a bit tricky, because, essentially, there is no end of song... :)

However, when I stop the song (local or radio), Audacious does not execute any command, which leaves the text file filled in with whatever played the last. That results in a false state that there is a song playing, when it isn't.

I hope I have managed to explain this properly and that you would be able to take this request into consideration.

Thank you very much in advance.

Cheer!

**Comment by radioactiveman** [comment]
This should be easy to implement. You don't need the format codes for this case, right?
Because if a song has been stopped, it is no longer possible to query its artist, title, file name etc.

**Comment by AbraxasDMG** [comment]
Hi. Thank you very much.

Yes, there is no need for format codes.

**Comment by radioactiveman** [comment]
Implemented with https://github.com/audacious-media-player/audacious-plugins/commit/b9e9511584084d7a70e87a8afc83c81a5775b308 for Audacious 4.4.
